src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java

```diff
@@ -65,9 +80,16 @@ public void updateEntity(DiscountRequestVO request, Discount discount) {
     }
 
     private void provideDiscountDependencies(DiscountRequestVO request, Discount discount) {
-        Vendor vendor = vendorRepository.findById(request.getVendorId()).orElse(null);
-        Category category = categoryRepository.findById(request.getCategoryId()).orElse(null);
+        var vendor = vendorRepository.findById(request.getVendorId())
+                .orElseThrow(() -> new EntityNotFoundException("Vendor not found"));
+        var category = categoryRepository.findById(request.getCategoryId())
+                .orElseThrow(() -> new EntityNotFoundException("Category not found"));
+
+        List<Location> locations = request.getLocationIds().stream()
+                .map(locationId -> locationRepository.findById(locationId).get())
```

Comment:
what will happen if there is no location with such id?

Reply:
added exception thrown if there is location with such id